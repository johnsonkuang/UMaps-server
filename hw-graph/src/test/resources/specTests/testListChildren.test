# Test ListChildren method
# Test whether a graph can list the children of a parent node

# Create graph
CreateGraph graph1

# Add parent and children nodes
AddNode graph1 parent
AddNode graph1 A
AddNode graph1 B
AddNode graph1 C
AddNode graph1 D
AddNode graph1 E

# Add edges between parent and all other nodes
AddEdge graph1 parent A e1
AddEdge graph1 parent B e2
AddEdge graph1 parent C e3
AddEdge graph1 parent D e4
AddEdge graph1 parent E e5

# List children of parent
ListChildren graph1 parent