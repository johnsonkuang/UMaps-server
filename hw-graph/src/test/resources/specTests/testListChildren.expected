# Test ListChildren method
# Test whether a graph can list the children of a parent node

# Create graph
created graph graph1

# Add parent and children nodes
added node parent to graph1
added node A to graph1
added node B to graph1
added node C to graph1
added node D to graph1
added node E to graph1

# Add edges between parent and all other nodes
added edge e1 from parent to A in graph1
added edge e2 from parent to B in graph1
added edge e3 from parent to C in graph1
added edge e4 from parent to D in graph1
added edge e5 from parent to E in graph1

# List children of parent
the children of parent in graph1 are: A(e1) B(e2) C(e3) D(e4) E(e5)